Docs: Fix Rails version number

vfonic opened the pull request.

vfonic commented:
Just realized there's exactly the same PR: https://github.com/rgeo/activerecord-postgis-adapter/pull/427

These version numbers might make more sense to have them in a markdown table so they don't take this much space.

seuros commented:
thank you, i just merged the other PR. 

I was going to do it early but i deleted the notification.

vfonic commented:
Formatting done.

I've removed 

> Gemfile:
> 
> ```ruby
> gem 'activerecord-postgis-adapter'
> ```

As it's again mentioned below under #Setup.

I've also removed 0.6.6 instructions as that's again mentioned in the linked 2.x README.

https://github.com/vfonic/activerecord-postgis-adapter/tree/patch-1?tab=readme-ov-file#version-compatibility
